assets/src/components/fromThemeOptions/fromThemeOptions.js

```diff
@@ -97,7 +99,9 @@ export function fromThemeOptions( WrappedComponent ) {
         return <WrappedComponent { ...ownProps }/>
       }
 
-      const field = getFieldFromTheme( theme, ownProps.metaKey );
+      const meta = wp.data.select( 'core/editor' ).getEditedPostAttribute( 'meta' );
```

Comment:
Is this needed here? I see the logic for merging the meta and the theme options is spread into a few parts, for example: we should possibly unify `withPostMeta` and `fromThemeOptions`, they look pretty dependant on each other.  

With this change we are retrieving the `meta` field in the `render` method of `fromThemeOptions`, a field should probably only need to know about its own value and not much else. So far, all the fields that use `fromThemeOptions` actually use `withPostMeta` in conjunction. Maybe you can use the `postMeta` prop in this HOC directly, since `compose` will pass the values right-to-left, and move the static methods from the outside to inside the classes, to avoid passing down the `meta` attribute to all of them?. I know `getFieldFromTheme` is exported separately and it calls `resolveDependency`, so I guess that will require some decoupling to do. Or some coupling actually, because now the `fromThemeOptions` file looks more like a set of utility function which rely heavily on the `meta` field. 

Anyway, I'm not sure, I think this probably needs to happen in some sort of `meta` handler on top of Campaign Sidebar and not in these HOCs which are used to decorate fields.

Reply:
> With this change we are retrieving the meta field in the render method of fromThemeOptions, a field should probably only need to know about its own value and not much else.

A field needs to know other values than its own from the meta, as it could depend on another field. The idea of the `fromThemeOptions` HOC is that it encapsulates this logic for reading from the theme. Technically would be cleaner to add `meta` as a prop, but that would involve having to provide the prop everywhere. We could instead use `withSelect` instead like in `withPostMeta`. I considered that, but it didn't seem worth the complexity at the moment, since we're only reading from the meta.

> Is this needed here? 
I just moved it up from here to have it called only once: https://github.com/greenpeace/planet4-plugin-gutenberg-blocks/pull/260/commits/589dee85fca1b47e3866934aaf376b9d6a6e6815#diff-e9a3f238758823bef04f80b5c8f492c1L28

Yeah currently `fromThemeOptions` and `withPostMeta` are somewhat coupled, though it is just `withThemeOptions` that depends on `withPostMeta`, not the other way round. Currently it's already used on its own for the theme dropdown, but I can imagine re-using it for other UI where we store things in `postMeta`, not necessarily related to the theme.

Maybe we could already decouple the theme from the meta storage too, on the other hand we already have 2 HOCs, so it might be more straightforward to include the coupling in `fromThemeOptions`, if we're not planning to use `fromThemeOptions` with other types of storage.

Reply:
> A field needs to know other values than its own from the meta, as it could depend on another field. The idea of the fromThemeOptions HOC is that it encapsulates this logic for reading from the theme.

Hmmm not sure about that, personally, I would make the CampaignSidebar smarter and the fields dumber, so maybe instead of supercharging the fields with `withPostMeta` and `fromThemeOptions`, I would move all the options & dependencies logic to the CampaignSidebar or even something on top of it.

Anyway, this looks good, I'm doing a round of manual tests locally just in case. 💯